components/dashboard/src/workspaces/CreateWorkspacePage.tsx

```diff
@@ -203,20 +199,20 @@ export function CreateWorkspacePage() {
             }
 
             try {
-                if (createWorkspaceMutation.isLoading || createWorkspaceMutation.isSuccess) {
+                if (createWorkspaceMutation.isStarting) {
                     console.log("Skipping duplicate createWorkspace call.");
                     return;
                 }
-                if (rememberOptions) {
-                    storeAutoStartOptions();
-                }
                 // we wait at least 5 secs
                 const timeout = new Promise((resolve) => setTimeout(resolve, 5000));
-                const result = await createWorkspaceMutation.mutateAsync({
+                const result = await createWorkspaceMutation.createWorkspace({
                     contextUrl: contextURL,
                     organizationId,
                     ...opts,
                 });
+                if (rememberOptions) {
+                    await storeAutoStartOptions();
```

Comment:
moving this to after calling and awaiting the createWorkspace call and actually awaiting it